### DISCUSSION TITLE
Can I use Serial (USB/FTDI) while powering the Upduino from the +5V pin?

### DISCUSSION THREAD

**Discussion by m00dawg** [tinyvision-ai-inc/UPduino-v3.0#26] [Q&A]
I plan on using the Upduino to do some prototyping with the Commander X16, a new retro 8-bit computer which provides expansion slots exposing the entire bus, power, etc. I was planning on powering the Upduino with this and would like to use the serial output over USB for debugging. Does the Upduino work like some microcontroller solutions (like say the Adafruit ItsyBitsy M4) where I can still using USB even if I'm directly supplying 5V?

As an aside, the X16 uses 5V logic so I'll be using bus transceivers in between the Upduino and the X16.

**Comment by vr2045** [comment]
Yes, but depends.
You have to be careful to not backdrive the USB 5V.

**Reply by m00dawg** [reply]
Ah got it, glad I asked! I do have an SPI to USB/FTDIO adapter. so I assume I can just hook it into the SPI pins and then just configure my application accordingly? I'll be starting with FemtoRV since it has serial I/O already and will go from there as I start to customize.

**Comment by vr2045** [comment]
Well, the ftdi is on the upduino already, not sure why you need another
one.
Sure, this should work.

**Reply by m00dawg** [reply]
Ah it was just a way to avoid the backfire situation you were referring to.

**Comment by vr2045** [comment]
If you're powering the upduino on your own, simply disconnect the PTC so it
doesn't back power the USB. You can still connect to the PC using the FTDI
on the board...

**Reply by m00dawg** [reply]
Aha got it ok. Dig through the schems and the HTML part layout tool (that thing if fantastic). Looks it's FB3.

**Comment by vr2045** [comment]
Yes, the ferrite bead was replaced with a PTC fuse so it doesn't fail when
the 5v is shorted.
Just desolder that and you don't need an alternate FTDI.

**Reply by m00dawg** [reply]
Fantastic, thanks for the help!
